frontend/app/src/scenes/Home/scenes/AddSubCategory/index.tsx

```diff
@@ -14,26 +14,35 @@ import {
   Typography,
 } from 'antd';
 import mainStyles from '../../styles.css';
-import axios, { AxiosResponse } from 'axios';
+import axios, { AxiosResponse, Method } from 'axios';
 import { handleApiError } from '../../../../services/util/errorHandler';
 
 const { Title, Text } = Typography;
 const { Option } = Select;
 
 class AddSubCategory extends React.Component<
-  RouteComponentProps,
+  RouteComponentProps<SubCategoryUrlParams>,
   AddSubCategoryStateProps
 > {
-  constructor(props: RouteComponentProps) {
+  subCategoryId: string | null;
+  componentType: 'edit' | 'add';
+  constructor(props: RouteComponentProps<SubCategoryUrlParams>) {
     super(props);
+    this.subCategoryId = this.props.match.params.subCategoryId;
+    this.componentType = 'add';
```

Comment:
I think you can decide the type from here

Reply:
oh yes